compositor_chromium/chromium_sys/build.rs

```
@@ -0,0 +1,163 @@
+    };
+
+    if !cef_root.exists() {
+        download_cef(&cef_root);
+    }
+
+    // target/debug or target/release directory
+    let target_path = out_dir
+        .parent()
+        .context("chromium_sys build directory not found")?
+        .parent()
+        .context("deps build directory not found")?
+        .parent()
+        .context("target build directory not found")?;
+    let bindings = prepare(&cef_root, &target_path)?;
+    bindings.write_to_file(PathBuf::from(".").join("src").join("bindings.rs"))?;
+
+    link(&cef_root, &target_path);
+
+    Ok(())
+}
+
+#[allow(unused_variables)]
+fn prepare(cef_root: &Path, target_path: &Path) -> Result<bindgen::Bindings> {
+    let bindings = bindgen::Builder::default()
+        .header("wrapper.h")
+        .clang_arg(format!("-I{}", cef_root.display().to_string()))
+        .parse_callbacks(Box::new(bindgen::CargoCallbacks))
+        .parse_callbacks(Box::new(RemoveCommentsCallback))
+        .generate()?;
+
+    #[cfg(target_os = "macos")]
+    {
+        let framework_out_path = target_path.join("Frameworks");
+        let _ = std::fs::create_dir_all(&framework_out_path);
+        let framework_path = PathBuf::from(cef_root)
+            .join("Release")
+            .join("Chromium Embedded Framework.framework");
+        let options = CopyOptions {
+            skip_exist: true,
+            ..Default::default()
+        };
+
+        dir::copy(framework_path, framework_out_path, &options)?;
+        dir::copy("resources", target_path, &options)?;
+    }
+
+    #[cfg(target_os = "linux")]
+    {
+        let options = CopyOptions {
+            skip_exist: true,
+            content_only: true,
+            ..Default::default()
+        };
+
+        let cef_root = PathBuf::from(cef_root);
+        let release_path = cef_root.join("Release");
+        let resources_path = cef_root.join("Resources");
+        let lib_path = target_path.join("lib");
+        let _ = fs::create_dir_all(&lib_path);
```

don't ignore an error